modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -517,10 +517,10 @@
   (sql-jdbc.execute/do-with-connection-with-options
    driver database nil
    (fn [^java.sql.Connection conn]
-     (with-open [stmt (.prepareStatement conn "show parameters like 'TIMEZONE';")
+     (with-open [stmt (.prepareStatement conn "show parameters like 'TIMEZONE' in user;")
```

Comment:
Rationale for changes to this method is further described in the PR description, Snowflake part.

Here are the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters#timezone) on timezone setting. So we have account, user and session timezone, where latter has a preference. User timezone is persistent across sessions. Session timezone is not (obvious, I know...). Hence I consider the user timezone (if not set, inherited from the account) as _the timezone that is used for connection without us overriding that (by setting report timezone)_ from the PR description.